src/plugins/navigation/public/plugin.tsx

```diff
@@ -66,10 +87,25 @@ export class NavigationPublicPlugin
     this.coreStart = core;
     this.depsStart = depsStart;
 
-    const { unifiedSearch, cloud } = depsStart;
+    const { unifiedSearch, cloud, security } = depsStart;
     const extensions = this.topNavMenuExtensionsRegistry.getAll();
     const chrome = core.chrome as InternalChromeStart;
 
+    if (security) {
+      this.userProfileOptOut$ = security.userProfiles.userProfileLoaded$.pipe(
+        skipWhile((loaded) => {
+          return !loaded;
+        }),
+        switchMap(() => {
+          return security.userProfiles.userProfile$ as Observable<UserProfileData>;
```

Comment:
Could this be simplified to just use `userProfile$`?

Maybe something like the following:

```js
if (security) {
      this.userProfileOptOut$ = security.userProfiles.userProfile$.pipe(
        map((profile) => {
          if (profile === null) return undefined;
          return (profile as UserProfileData)?.userSettings?.solutionNavOptOut;
        }),
        distinctUntilChanged()
      );
}
```

Reply:
This goes back to my previous comment.

My intention is to first wait for the user to be fetched before continuing with the flow. This avoids (1) flickers in the UI (we show a black header and quickly after a white one) and (2) downstream consumer to receive first an `undefined` value that could mean `false` and then a few ms later a `true`. They compute logic for nothing because of that.